### PR TITLE
adding spain state data

### PR DESCRIPTION
**Pull request by osvyg**
Resolves https://github.com/lazd/coronadatascraper/issues/81

I am making the assumption that the data is from https://opendata.arcgis.com/datasets/48fac2d7de0f43f9af938852e3748845_0.csv 

Also, I am calling them "state" but they might be "regions"

### CODE REVIEW

File: scrapers.js [hunk 1]
```diff
@@ -1686,6 +1686,32 @@ let scrapers = [
...
+            cases,
+            deaths,
+            recovered,
+          };
+          if (rules.isAcceptable(data, null, this._reject)) {
+            states.push(data);
+          }
+        }
+      }
+      return states;
```
Comment (lazd) [inline scrapers.js]:
```suggestion
      // Add data for ESP
      states.push(transform.sumData(states);

      return states;
```

File: scrapers.js [hunk 2]
```diff
@@ -1686,6 +1686,32 @@ let scrapers = [
       };
     }
   },
+  {
+    country: 'ESP',
+    url: 'https://opendata.arcgis.com/datasets/48fac2d7de0f43f9af938852e3748845_0.csv',
```
Comment (lazd) [inline scrapers.js]: Make this the authoritative data source for ESP:

```suggestion
    url: 'https://opendata.arcgis.com/datasets/48fac2d7de0f43f9af938852e3748845_0.csv',
    priority: 1,
```

File: scrapers.js [hunk 3]
```diff
@@ -1686,6 +1686,36 @@ let scrapers = [
...
+            deaths,
+            recovered,
+          };
+          if (rules.isAcceptable(data, null, this._reject)) {
+            states.push(data);
+          }
+        }
+      }
+      // Add data for ESP
+      states.push(transform.sumData(states);
```
Comment (lazd) [inline scrapers.js]:
```suggestion
      states.push(transform.sumData(states));
```

My bad!

File: scrapers.js [hunk 4]
```diff
@@ -1686,6 +1686,36 @@ let scrapers = [
...
+    priority: 1,
+    scraper: async function() {
+      let data = await fetch.csv(this.url);
+      let states = [];
+      for (let row of data) {
+        let state = row.Texto !== undefined ? parse.string(row.Texto) : "";
+        let cases = row.TotalConfirmados !== undefined ? parse.number(row.TotalConfirmados) : 0;
+        let deaths = row.TotalFallecidos !== undefined ? parse.number(row.TotalFallecidos) : 0;
+        let recovered = row.TotalRecuperados !== undefined ? parse.number(row.TotalRecuperados) : 0;
+        if(state !== "") {
```
Comment (lazd) [inline scrapers.js]: So what are rows with no state? Are they unassigned cases?

Comment (osvyg) [reply scrapers.js]: They are always there in the data, I was mostly making sure the column "Texto" exists.

**Pull request merged**